ntp-daemon/src/keyexchange.rs

```diff
@@ -36,6 +44,48 @@ pub(crate) async fn key_exchange(
     BoundKeyExchangeClient::new(socket, server_name, config)?.await
 }
 
+pub async fn key_exchange_server(
+    address: impl ToSocketAddrs,
+    certificate_chain: Vec<Certificate>,
+    key_der: PrivateKey,
+) -> std::io::Result<()> {
+    use std::io;
+
+    let listener = TcpListener::bind(address).await?;
+
+    let provider = KeySetProvider::new(8);
+
+    let mut config = rustls::ServerConfig::builder()
+        .with_safe_defaults()
+        .with_no_client_auth()
+        .with_single_cert(certificate_chain, key_der)
+        .map_err(|err| io::Error::new(io::ErrorKind::InvalidInput, err))?;
+
+    config.alpn_protocols.clear();
+    config.alpn_protocols.push(b"ntske/1".to_vec());
+
+    let config = Arc::new(config);
+
+    loop {
+        let (stream, peer_addr) = listener.accept().await?;
+        let config = config.clone();
+        let keyset = provider.get();
+
+        let fut = async move {
+            BoundKeyExchangeServer::run(stream, config, keyset)
+                .await
+                .map_err(|ke_error| std::io::Error::new(std::io::ErrorKind::Other, ke_error))
+        };
+
+        tokio::spawn(async move {
+            match fut.await {
+                Err(err) => tracing::error!(?err, ?peer_addr, "NTS KE failed"),
+                Ok(()) => tracing::info!(?peer_addr, "NTS KE completed"),
```

Comment:
Given how often both of these will occur on a typical server, and the fact that errors are quite often not errors on our side (it could just be the client closing the connection earlier than we expect or something like that), I would suggest significantly lowering the importance of these messages (debug for Ok, and debug or info for the error case)

Reply:
good call